partition.go

```diff
@@ -0,0 +1,141 @@
+package schema
+
+import (
+	"encoding/binary"
+	"hash/fnv"
+	"sort"
+
+	"github.com/cespare/xxhash"
+	jump "github.com/dgryski/go-jump"
+)
+
+type PartitionByMethod uint8
+
+const (
+	PartitionByOrg PartitionByMethod = iota
+	PartitionBySeries
+	PartitionBySeriesWithTags
+	PartitionBySeriesWithTagsFnv
+)
+
+func (m *MetricData) PartitionID(method PartitionByMethod, partitions int32) (int32, error) {
+	var partition int32
+
+	switch method {
+	case PartitionByOrg:
+		h := fnv.New32a()
+		err := binary.Write(h, binary.LittleEndian, uint32(m.OrgId))
+		if err != nil {
+			return 0, err
+		}
+		partition = int32(h.Sum32()) % partitions
+		if partition < 0 {
+			partition = -partition
+		}
+	case PartitionBySeries:
+		h := fnv.New32a()
+		h.Write([]byte(m.Name))
+		partition = int32(h.Sum32()) % partitions
+		if partition < 0 {
+			partition = -partition
+		}
+	case PartitionBySeriesWithTags:
+		h := xxhash.New()
+		h.WriteString(m.Name)
+		sort.Strings(m.Tags)
+		for _, t := range m.Tags {
+			if len(t) >= 5 && t[:5] == "name=" {
+				continue
+			}
+
+			h.WriteString(";")
+			h.WriteString(t)
+		}
+		partition = jump.Hash(h.Sum64(), int(partitions))
+	case PartitionBySeriesWithTagsFnv:
+		h := fnv.New32a()
+		h.Write([]byte(m.Name))
+		sort.Strings(m.Tags)
```

Comment:
Same comment as `PartitionBySeriesWithTags`

Reply:
most likely not worth trying to shave a couple instructions